### PR TITLE
demo notebook for HVG experimental API

### LINKED ISSUE #532: Add HVGs notebook
- Please follow [editorial guidelines](https://github.com/chanzuckerberg/cellxgene-census/blob/main/docs/cell_census_notebook_guidelines.md). 
- Add notebook in new folder `experimental` under `api/python/notebooks`
- And please add symlink to `docs/notebooks`

### LINKED ISSUE #535: pre-commit reporting builder errors
not sure if this a pre-commit config issue, but the warnings look real.

to reproduce, run `pre-commit run -a -v`

```
mypy-tools...............................................................Failed
- hook id: mypy
- duration: 1.97s
- exit code: 1

tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/experiment_builder.py:96: error: Too many arguments for "ExperimentSpecification"  [call-arg]
tools/cellxgene_census_builder/tests/conftest.py:25: error: Too many arguments for "Organism"  [call-arg]
tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/__main__.py:18: error: Unexpected keyword argument "working_dir" for "CensusBuildArgs"  [call-arg]
tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/__main__.py:18: error: Unexpected keyword argument "config" for "CensusBuildArgs"  [call-arg]
tools/cellxgene_census_builder/src/cellxgene_census_builder/__main__.py:36: error: Unexpected keyword argument "working_dir" for "CensusBuildArgs"  [call-arg]
tools/cellxgene_census_builder/src/cellxgene_census_builder/__main__.py:36: error: Unexpected keyword argument "config" for "CensusBuildArgs"  [call-arg]
tools/cellxgene_census_builder/src/cellxgene_census_builder/__main__.py:36: error: Unexpected keyword argument "state" for "CensusBuildArgs"  [call-arg]
Found 7 errors in 4 files (checked 41 source files)
```

### PR DESCRIPTION
**Pull request by bkmartinjr**
Changes:
* created experimental API notebooks folder and moved pytorch notebook into experimental API folder
* added HVG notebook to experimental folder
* cleaned up docs/notebooks symlinks (removed dead links, added missing link)
* pre-commit fixes captured in #539 

Fixes #532 
Fixes #535

*Note to reviewers:*  the pre-commit and builder lint issues are addressed in #539.  I preemptively merged that PR into this branch, but please refer to #539 for details on those changes (and feel free to leave f/b on it in either PR - will address issues here).

### CODE REVIEW

File: api/python/notebooks/experimental/highly_variable_genes.ipynb [hunk 1]
```diff
@@ -0,0 +1,945 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Experimental Highly Variable Genes API\n",
+    "\n",
+    "This tutorial describes use of the `cellxgene_census.experimenta.pp` API for finding highly variable genes (HVGs) in the Census. The HVG algorithm implements the ranked normalized variance method `seurat_v3` described in [`scanpy.pp.highly_variable_genes](https://scanpy.readthedocs.io/en/stable/generated/scanpy.pp.highly_variable_genes.html#scanpy.pp.highly_variable_genes).\n",
```
Comment (ebezzi) [inline api/python/notebooks/experimental/highly_variable_genes.ipynb:10]:
```suggestion
    "This tutorial describes use of the `cellxgene_census.experiments.pp` API for finding highly variable genes (HVGs) in the Census. The HVG algorithm implements the ranked normalized variance method `seurat_v3` described in [`scanpy.pp.highly_variable_genes](https://scanpy.readthedocs.io/en/stable/generated/scanpy.pp.highly_variable_genes.html#scanpy.pp.highly_variable_genes).\n",
```

File: api/python/notebooks/experimental/highly_variable_genes.ipynb [hunk 2]
```diff
@@ -0,0 +1,945 @@
...
+    "\n",
+    "There are two API available:\n",
+    "* `get_highly_variable_genes()` - high level function which accepts arguments similar to `cellxgene_census.get_anndata()`, and returns annotations for each `var` feature in a Pandas DataFrame.\n",
+    "* `highly_variable_genes()` - lower level function which accepts a `tiledbsoma.ExperimentAxisQuery` and returns the same result.\n",
+    "\n",
+    "Both functions accept common arguments to control ranking, with argument semantics matching the ScanPY API:\n",
```
Comment (ebezzi) [inline api/python/notebooks/experimental/highly_variable_genes.ipynb:16]:
```suggestion
    "Both functions accept common arguments to control ranking, with argument semantics matching the Scanpy API:\n",
```
This is the spelling used on their website

File: api/python/notebooks/experimental/highly_variable_genes.ipynb [hunk 3]
```diff
@@ -0,0 +1,945 @@
...
+    "* `n_top_genes` - number of genes to rank\n",
```
Comment (ebezzi) [inline api/python/notebooks/experimental/highly_variable_genes.ipynb:17]:
```suggestion
    "* `n_top_genes` - number of genes to rank.\n",
```

File: api/python/notebooks/experimental/highly_variable_genes.ipynb (hunk 1 above)
Comment (ebezzi) [reply api/python/notebooks/experimental/highly_variable_genes.ipynb:10]:
```suggestion
    "This tutorial describes use of the `cellxgene_census.experimenta.pp` API for finding highly variable genes (HVGs) in the Census. The HVG algorithm implements the ranked normalized variance method `seurat_v3` described in [scanpy.pp.highly_variable_genes](https://scanpy.readthedocs.io/en/stable/generated/scanpy.pp.highly_variable_genes.html#scanpy.pp.highly_variable_genes).\n",
```

Comment (ebezzi) [reply api/python/notebooks/experimental/highly_variable_genes.ipynb:10]: Link is broken otherwise.

**Review by ebezzi — CHANGES_REQUESTED** [review]
A few typos/corrections but otherwise LGTM.

**Review by ebezzi — APPROVED** [review]

**Review by pablo-gar — APPROVED** [review]
Looks awesome! Thanks for cleaning up the symlinks (errors and removal of unused links)

**Review by atolopko-czi — APPROVED** [review]
LGTM

**Pull request merged**